### PR TITLE
feat: include the commit deployed in PR comments

### PR DESCRIPTION
**Pull request by RyanGWU82**
Based on a suggestion from Kanad, I've updated PR messages to add the commit that was deployed. If you do multiple `git push`es in succession, and it takes a while for the app to redeploy, it's sometimes hard to tell exactly which commit is currently on Heroku.

Example of the new message:

Screen Shot 2022-05-17 at 2 18 15 PM

### CODE REVIEW

File: src/git.js [hunk 1]
```diff
@@ -30,6 +31,28 @@ module.exports.refExists = ref => {
   return result.status === 0;
 };
 
+/* Returns the SHA-1 hash of the latest commit in the given ref. */
+module.exports.shaForRef = ref => {
```
Comment (kanadgupta) [inline src/git.js:35]: FYI most of these are available as env variables in GitHub Actions runners: https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables

But if this approach appears to work just fine 🤷🏽

**Review by kanadgupta — APPROVED** [review]

File: src/git.js (hunk 1 above)
Comment (RyanGWU82) [reply src/git.js]: I was able to remove `shaForRef()` but the commit message isn't in the environment or the context object, so I kept that one in here.

**Review by erunion — APPROVED** [review]

File: src/comments.js [hunk 2]
```diff
@@ -42,24 +42,44 @@ function formatComment(options) {
...
+
+  const owner = github.context.payload.repository.owner.login;
+  const repo = github.context.payload.repository.name;
+  const prNumber = parseInt(github.context.payload.number, 10);
+  const sha = github.context.sha;
+
+  if (owner && repo && prNumber && sha && message) {
+    const commitLink = `https://github.com/${owner}/${repo}/pull/${prNumber}/commits/${sha}`;
+    const commitText = `\`${sha.substring(0, 8)}\` ${message.split('\n')[0]}`;
+    links.push(`:rocket: **Deployed commit:** [${commitText}](${commitLink})`);
```
Comment (kanadgupta) [inline src/comments.js:56]: Ooooh I was wondering why you were constructing the commit links by hand like this... and I just learned that these two links result in different pages!
```
https://github.com/readmeio/heroku-review-app-action/commit/b0814f6230e49037186869dc3a5ceddbf8e47863
https://github.com/readmeio/heroku-review-app-action/pull/4/commits/b0814f6230e49037186869dc3a5ceddbf8e47863
```
Cool with this approach, linking to the commit within the PR is a lot more useful 👍🏽

**Review by kanadgupta — APPROVED** [review]

**Comment by kanadgupta** [comment]
Hey @RyanGWU82! Just coming back to this, what are your thoughts on having the Action comment once when the app deploys and have it make subsequent edits to that comment with subsequent redeploys? With the changes in this PR I feel like that'll look a lot cleaner since the user will know which commit has been deployed!

File: src/comments.js [hunk 3]
```diff
@@ -1,74 +1,157 @@
...
+      issues: {
+        createComment: () => {},
+        updateComment: () => {},
+        listComments: () => {},
+      },
+    },
+  };
+}
+
+let octokit;
```
Comment (erunion) [inline src/comments.js:30]: I'm surprised that this being here isn't throwing any parse or ESLint errors on line 55 where you do `const octokit`.

**Comment by RyanGWU82** [comment]
@kanadgupta @erunion: This is ready for re-review. Based on the feedback in [today's Slack thread](https://readmeio.slack.com/archives/C5B798RFZ/p1653491855171139), I've updated this to post a single message when the review app is opened, and then append log entries to that when the app is redeployed or shut down. Here's how it looks in action:

Screen Shot 2022-05-25 at 4 02 52 PM

I also fixed the issue where it would show a merge commit message instead of the message from the actual commit in this branch.

File: src/comments.js (hunk 3 above)
Comment (RyanGWU82) [reply src/comments.js:30]: So am I 🤷🏼  I'll rename this to `memoizedOctokit` (unless you've got a better idea?)

Comment (erunion) [reply src/comments.js:30]: renaming it should be good enough

**Review by erunion — APPROVED** [review]
:nice:

**Pull request merged**